### PR TITLE
Find VMware vCenter connected to BackupServer associated with need BackupRersource...

### PR DESCRIPTION
**Pull request by ykohut**
…, while Backup Job is added for Virtual Server.

### CODE REVIEW

File: schedule_create.rb [hunk 1]
```diff
@@ -122,19 +133,37 @@ def add_vm_params(vm_to_add_ref)
...
-    hierarchy_root_id =
-      uid_to_identifier(
-        :HierarchyRoot,
-        api_get(
-          build_query(:hierarchyroot, { uniqueid: "\"#{vcenter_instance_uuid(virtual_server)}\"" }, entities: false)
-        ).dig(:QueryResult, :Refs, :Ref, :UID)
-      )
-
-    return unless hierarchy_root_id
+    hierarchyroots =
```
Comment (hasrthur) [inline schedule_create.rb]:
```suggestion
    hierarchy_roots =
```

**Review by hasrthur — APPROVED** [review]

File: schedule_create.rb [hunk 2]
```diff
@@ -84,6 +84,17 @@ def backup_repository_uid
...
 
+  def backup_server_id
+    backup_repository_ref =
+      api_get("repositories/#{uid_to_identifier(:Repository, backup_repository_uid)}")
+        .dig(:EntityRef, :Links, :Link)
+
+    return unless backup_repository_ref
+
+    backup_repository_ref.detect { |hash| hash[:Type] == 'BackupServerReference' }[:Href]
+      .rpartition('/')[2] || ""
```
Comment (nahaylo) [inline schedule_create.rb]: `backup_repository_ref.detect { |hash| hash[:Type] == 'BackupServerReference' }.fetch(:Href, "").rpartition('/').last`

File: schedule_create.rb [hunk 3]
```diff
@@ -122,19 +133,37 @@ def add_vm_params(vm_to_add_ref)
...
-        ).dig(:QueryResult, :Refs, :Ref, :UID)
-      )
-
-    return unless hierarchy_root_id
+    hierarchyroots =
+      api_get(
+        build_query(:hierarchyroot, { uniqueid: "\"#{vcenter_instance_uuid(virtual_server)}\"" }, entities: false)
+      ).dig(:QueryResult, :Refs, :Ref)
+
+    return unless hierarchyroots
```
Comment (nahaylo) [inline schedule_create.rb]: add method
`
def hierarchy_roots
  @hierarchy_roots ||= api_get()....
end
`

and then here simple

`return unless hierarchy_roots`

File: schedule_create.rb [hunk 4]
```diff
@@ -122,19 +133,37 @@ def add_vm_params(vm_to_add_ref)
...
+        api_get("hierarchyRoots/#{hierarchy_root_id}").dig(:EntityRef, :Links, :Link)
+      next unless backup_server_href
+
+      backup_server_url =
+        backup_server_href.detect { |hash| hash[:Type] == 'BackupServerReference' }[:Href]
+      next unless backup_server_url
+
+      # BackupServer ID of VM hierarchyroot is equal to BackupServer ID of BackupRepository
+      if backup_server_url
+           .rpartition('/')[2]
```
Comment (nahaylo) [inline schedule_create.rb]: `rpartition('/').last`

File: schedule_create.rb (hunk 1 above)
Comment (ykohut) [reply schedule_create.rb]: Done.

File: schedule_create.rb (hunk 4 above)
Comment (ykohut) [reply schedule_create.rb]: Done.

File: schedule_create.rb (hunk 3 above)
Comment (ykohut) [reply schedule_create.rb]: Rejected.

**Review by nahaylo — APPROVED** [review]

**Pull request merged**